### DISCUSSION TITLE
For 3D object, is it possible to make the object anti aliasing?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1055] [Q&A]
Hi:

Seems like FXGL 11.16 supports 3D object
and we have tested it with cone
but when we place the cone in the game it seems like the border is aliasing
like this
<img width="179" alt="Screen Shot 2021-05-28 at 9 45 49 AM" src="https://user-images.githubusercontent.com/5525436/119917633-83f97f80-bf99-11eb-8667-60cf0a491eeb.png">

Is there any way we could make the border anti-aliasing?

Thanks

**Comment by chengenzhao** [comment]
Is it a good idea to place a sub scene to contain shape 3d objects?
and set anti aliasing to the sub scene?

**Reply by AlmasB** [reply]
I am not sure about anti-aliasing, I think it needs to be baked into mesh info, particularly the face smoothing. You can try producing a 3D model with correct smoothing information in a 3D modelling software and export as .obj. Finally, load via: `getAssetLoader().loadModel("file.obj");`

**Reply by chengenzhao** [reply]
Good Morning
Using proper smoothing info. could be a way to do that
but the cone is purely produced by code
and we have JavaFX project suggest a solution which is using SceneAntialiasing parameter to reduce the alias 
https://openjfx.io/javadoc/16/javafx.graphics/javafx/scene/SceneAntialiasing.html
SceneAntialiasing.BALANCED
that could enable the anti aliasing feature of OpengGL or D3D or Metal soon
if we don't want the whole scene to enable that (for performance consideration) we could use subscene to do that
like this
![A13A9B00C734930EB3D74B7C22DF8970](https://user-images.githubusercontent.com/5525436/120047505-55d27900-c047-11eb-914f-d46f064ffd24.png)
and our friend made a mc skin viewer using this feature
https://github.com/InfinityStudio/minecraft-jfx-skin/

**Reply by chengenzhao** [reply]
Code may similar to this:
https://github.com/InfinityStudio/minecraft-jfx-skin/blob/master/src/moe/mickey/minecraft/skin/fx/SkinCanvas.java

and for WolFX, this feature maybe helpful?

**Reply by AlmasB** [reply]
Thanks, will check it out later.
